extensions/vscode/lsp-server/src/common.ts

```diff
@@ -140,7 +100,12 @@ export function prepareConnection(connection: Connection, isWeb: Boolean) {
 
     // Create lsp handlers
     connection.onInitialize(async (params: InitializeParams): Promise<InitializeResult | ResponseError<InitializeError>> => {
-        return await getResponse(createLspJson('initialize', params));
+        lspModule = await wakeLspModule(connection);
+        lspModule?._instantiateServer();
+
+        let response: InitializeResult | ResponseError<InitializeError> = await getResponse(createLspJson('initialize', params));
+        // TODO: impl & test invalid stdlib
```

Comment:
I think I still need to do this, unclear if VSCode will extract the notification for us or if it needs to be manually extracted as it was before

Reply:
I'm not sure what this code is doing, could you explain it a a bit more?

Reply:
`await getResponse(createLspJson('initialize', params));` is just exchanging a message with the lsp server.

`response` _may_ have a `notification` set if the stdLib being sent to the server isn't valid. In that case we need to show the notification to the user.

In the old code we manually extracted the notification and replayed it to the user. My comment here is just that I need to test that out and see if we actually need to do the replay or if that was some weird quirk of the hack. In _theory_, I'd expect VSCode to just handle the notification automatically but I need to confirm

Reply:
Okay, I just tested this and it actually just works without the manual replay. So we are actually removing two hacks in the PR :)